core/src/main/scala/kafka/server/epoch/LeaderEpochFileCache.scala

```diff
@@ -160,6 +160,18 @@ class LeaderEpochFileCache(topicPartition: TopicPartition,
     }
   }
 
+  def findPreviousEpoch(epoch: Int): Option[Int] = {
+    inReadLock(lock) {
+      Option(epochs.lowerKey(epoch))
```

Comment:
The `2.8x-tiered-storage` branch has this code

```
Option(epochs.lowerEntry(epoch)).map(_.getKey)
```

I made this change to call `lowerKey` directly instead of `lowerEntry`, and also suggested this change to Satish on his PR to trunk. He too made the code change on that PR - https://github.com/apache/kafka/pull/11390/commits/8144b29dd13d04d543ff94fb7a49b191a5c10195#diff-4ed1f35d9b2e9e01c17e9a75bf7fa7ab4fb03dfdc8a59f205f00920a35b9c69a